src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -3306,11 +3326,14 @@ def create_image_version(cmd, resource_group_name, gallery_name, gallery_image_n
             raise CLIError('usage error: Please provide --managed-image')
         image_version = ImageVersion(publishing_profile=profile, location=location, tags=(tags or {}))
 
-    return client.gallery_image_versions.create_or_update(resource_group_name=resource_group_name,
-                                                          gallery_name=gallery_name,
-                                                          gallery_image_name=gallery_image_name,
-                                                          gallery_image_version_name=gallery_image_version,
-                                                          gallery_image_version=image_version)
+    return client.gallery_image_versions.begin_create_or_update(
+        resource_group_name=resource_group_name,
+        gallery_name=gallery_name,
+        gallery_image_name=gallery_image_name,
+        gallery_image_version_name=gallery_image_version,
+        gallery_image_version=image_version,
+        headers={'x-ms-authorization-auxiliary': external_bearer_token}
```

Comment:
This is only a temporary workaround which is against the [`TokenCredential.get_token`](https://github.com/Azure/azure-sdk-for-python/blob/7c4d042141d512e212fb557808aa0a2ae8c02334/sdk/core/azure-core/azure/core/credentials.py#L16-L25) pattern.

`x-ms-authorization-auxiliary` should eventually be supported by SDK clients as something like `external_credential` (https://github.com/Azure/azure-sdk-for-python/issues/8313).